packages/server/routes/api/auth.js

```diff
@@ -283,4 +296,168 @@ router.post("/token/revoke", async (request, response) => {
   }
 });
 
+// TODO: Test Route
+router.post(
+  "/forgot-password",
+  forgotPasswordValidator,
+  async (request, response) => {
+    // Express validators for email checks
+    const result = validationResult(request);
+
+    if (!result.isEmpty()) {
+      return response.status(400).json({
+        status: "fail",
+        data: result.array(),
+      });
+    }
+
+    const data = matchedData(request);
+
+    try {
+      const email = data.email;
+
+      // Check if the user has an existing account.
+      const verifyEmailResult = await AuthController.verifyUserEmail(email);
+      if (!verifyEmailResult) {
+        return response.status(400).json({
+          status: "fail",
+          data: {
+            message: "User with given email does not exist.",
+          },
+        });
+      }
+
+      // Grab the userID associated with the provided email.
+      const userId = await UserController.getUserIdByEmail(email);
+
+      // Check if the user's account is a google OAuth Account.
+      const isGoogleAccountResult = await AuthController.isGoogleAccount(
+        userId
+      );
+
+      if (isGoogleAccountResult) {
+        return response.status(400).json({
+          status: "fail",
+          data: {
+            message: "Please login using Google.",
+          },
+        });
+      }
+
+      // Generate the JWT password reset token
+      const passwordResetToken =
+        TokenGenerator.generateResetPasswordToken(userId);
+      if (!passwordResetToken) {
+        return response.status(400).json({
+          status: "fail",
+          data: {
+            message: "Could not generate a token from the provided userID.",
+          },
+        });
+      }
+
+      // Use template literal to embed the token as a query inside the link (Note: this is a testing link since the app is not hosted yet.)
+      const link = `http://localhost:5050/reset-password?token=${passwordResetToken}`;
+
+      // Uncomment the line below when the app is offically launched.
+      // const link = `https://icebreaksea.com/account/reset-password?token=${passwordResetToken}`;
```

Comment:
Like I said in my other comment about duplicate code in your sending email functions, create the link within the `sendPasswordResetEmail()` function instead.